graph/iterator/resolver.go

```diff
@@ -0,0 +1,161 @@
+type Resolver struct {
+	qs     graph.QuadStore
+	uid    uint64
+	tags   graph.Tagger
+	nodes  []quad.Value
+	index  int
+	err    error
+	result graph.Value
+}
+
+// Creates a new Resolver iterator.
+func NewResolver(qs graph.QuadStore, nodes ...quad.Value) *Resolver {
+	it := &Resolver{
+		uid:   NextUID(),
+		qs:    qs,
+		nodes: make([]quad.Value, 0, 20),
+	}
+	// Enforce uniqueness
+	unique := make(map[quad.Value]bool, 20)
+	for _, node := range nodes {
+		if _, ok := unique[node]; !ok {
+			unique[node] = true
+			it.nodes = append(it.nodes, node)
+		}
+	}
+	return it
+}
+
+func (it *Resolver) UID() uint64 {
+	return it.uid
+}
+
+func (it *Resolver) Reset() {
+	it.index = 0
+	it.err = nil
+	it.result = nil
+}
+
+func (it *Resolver) Close() error {
+	return nil
+}
+
+func (it *Resolver) Tagger() *graph.Tagger {
+	return &it.tags
+}
+
+func (it *Resolver) TagResults(dst map[string]graph.Value) {
+	it.tags.TagResult(dst, it.Result())
+}
+
+func (it *Resolver) Clone() graph.Iterator {
+	out := NewResolver(it.qs, it.nodes...)
+	out.tags.CopyFrom(it)
+	return out
+}
+
+func (it *Resolver) String() string {
+	return fmt.Sprintf("Resolver(%v)", it.nodes)
+}
+
+// Register this iterator as a Resolver iterator.
+func (it *Resolver) Type() graph.Type { return graph.Fixed }
+
+// Check if the passed value is equal to one of the nodes stored in the iterator.
+func (it *Resolver) Contains(ctx context.Context, value graph.Value) bool {
+	graph.ContainsLogIn(it, value)
+	for _, node := range it.nodes {
+		if it.qs.ValueOf(node) == value {
+			return graph.ContainsLogOut(it, value, true)
+		}
+	}
+	return graph.ContainsLogOut(it, value, false)
+}
+
+// Next advances the iterator.
+func (it *Resolver) Next(ctx context.Context) bool {
+	graph.NextLogIn(it)
+	if it.index == len(it.nodes) {
+		it.result = nil
+		return graph.NextLogOut(it, false)
+	}
+	node := it.nodes[it.index]
+	value := it.qs.ValueOf(node)
+	if value == nil {
+		it.result = nil
+		it.err  = fmt.Errorf("not found: %v", node)
+		return false
+	}
+	it.result = value
+	it.index++
+	return graph.NextLogOut(it, true)
+}
+
+func (it *Resolver) Err() error {
+	return it.err
+}
+
+func (it *Resolver) Result() graph.Value {
+	return it.result
+}
+
+func (it *Resolver) NextPath(ctx context.Context) bool {
+	return false
+}
+
+func (it *Resolver) SubIterators() []graph.Iterator {
+	return nil
+}
+
+// Returns a Null iterator if it's empty so that upstream iterators can optimize it
+// away, otherwise there is no optimization.
+func (it *Resolver) Optimize() (graph.Iterator, bool) {
+	if len(it.nodes) == 0 {
+		return NewNull(), true
+	}
+	return it, false
+}
+
+// Size is the number of m stored.
+func (it *Resolver) Size() (int64, bool) {
+	return int64(len(it.nodes)), true
+}
+
+func (it *Resolver) Stats() graph.IteratorStats {
+	s, exact := it.Size()
+	return graph.IteratorStats{
+		// Lookup cost is size of set
+		ContainsCost: s - int64(it.index),
```

Comment:
```suggestion
		ContainsCost: s,
```
The assumption is that `Stats` is stateless.